packages/codegen-ui-react/lib/react-studio-template-renderer.ts

```diff
@@ -531,11 +531,12 @@ export abstract class ReactStudioTemplateRenderer extends StudioTemplateRenderer
     }
 
     // remove overrides from rest of props
+    const hasVariant = isStudioComponentWithVariants(component);
     elements.push(
       factory.createBindingElement(
         undefined,
-        factory.createIdentifier('overrides'),
-        factory.createIdentifier('overridesProp'),
+        hasVariant ? factory.createIdentifier('overrides') : undefined,
+        hasVariant ? factory.createIdentifier('overridesProp') : factory.createIdentifier('overrides'),
```

Comment:
Can this be ? ` factory.createIdentifier(hasVariant? 'overridesProp' : 'overrides')`

Reply:
Yeah, nice catch. I'll update.